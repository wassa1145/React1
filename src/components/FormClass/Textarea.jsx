import React, { Component } from 'react';

export class Textarea extends Component {
	render() {
		return <textarea rows="6" className="form__textarea" value={this.props.value} onChange={this.props.change}/>
  }
}
