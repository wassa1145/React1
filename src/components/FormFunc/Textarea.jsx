export const Textarea = (props) => {
	return <textarea rows="6" className="form__textarea" value={props.value} onChange={props.change}/> 
}
