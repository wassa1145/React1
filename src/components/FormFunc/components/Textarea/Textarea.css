.form__textarea {
  width: 100%;
  background-color: #fff;
  margin-right: 20px !important;
}
