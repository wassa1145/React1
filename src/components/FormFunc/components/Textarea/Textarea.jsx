import React from 'react';
import './Textarea.css';

export const Textarea = (props) => {
  return (
    <textarea
      rows="6"
      className="form__textarea"
      placeholder={props.placeholder}
      value={props.message}
      onChange={props.change}
      data-testid="textarea"
    />
  );
};
