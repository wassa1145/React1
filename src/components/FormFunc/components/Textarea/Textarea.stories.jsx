import React from 'react';

import { Textarea } from './Textarea';

export default {
  title: 'MyComponents/Textarea',
  component: Textarea,
};

const Template = (args) => <Textarea {...args} />;

export const Placeholder = Template.bind({});
Placeholder.args = {
  placeholder: 'Введите сообщение',
};

export const Filling = Template.bind({});
Filling.args = {
  placeholder: 'Введите сообщение',
  message: 'Новое сообщение',
};
