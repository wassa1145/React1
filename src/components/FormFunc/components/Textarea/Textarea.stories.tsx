import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Textarea } from './Textarea';

export default {
  title: 'MyComponents/Textarea',
  component: Textarea,
} as ComponentMeta<typeof Textarea>;

const Template: ComponentStory<typeof Textarea> = (args) => (
  <Textarea {...args} />
);

export const Placeholder = Template.bind({});
Placeholder.args = {
  placeholder: 'Введите сообщение',
};

export const Filling = Template.bind({});
Filling.args = {
  placeholder: 'Введите сообщение',
  message: 'Новое сообщение',
};
