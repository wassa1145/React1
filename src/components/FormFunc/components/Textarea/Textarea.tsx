import React, { FC } from 'react';
import TextField from '@mui/material/TextField';
import './Textarea.css';

interface TextareaProps {
  placeholder: string;
  message?: string;
  textFieldRef?: React.RefObject<HTMLInputElement>;
  change: (event: React.ChangeEvent<HTMLTextAreaElement>) => void;
}
export const Textarea: FC<TextareaProps> = (props) => {
  return (
    <TextField
      variant="outlined"
      className="form__textarea"
      multiline
      rows={4}
      placeholder={props.placeholder}
      value={props.message}
      onChange={props.change}
      inputRef={props.textFieldRef}
      data-testid="textarea"
    />
  );
};
