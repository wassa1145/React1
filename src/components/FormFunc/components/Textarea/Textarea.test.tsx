import React from 'react';
import { Textarea } from './Textarea';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

beforeEach(() => {
  render(<Textarea change={jest.fn()} placeholder="Введите сообщение" />);
});

describe('Textarea', () => {
  it('render component', () => {
    expect(screen.getByTestId('textarea')).toBeInTheDocument();
  });

  it('render with snapshot', () => {
    const { asFragment } = render(
      <Textarea change={jest.fn()} message="" placeholder="Введите сообщение" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('render component', () => {
    expect(screen.getByTestId('textarea')).toBeInTheDocument();
  });

  it('input text', async () => {
    await userEvent.type(screen.getByRole('textbox'), 'new message');
    expect(screen.getByRole('textbox')).toHaveValue('new message');
  });
});
