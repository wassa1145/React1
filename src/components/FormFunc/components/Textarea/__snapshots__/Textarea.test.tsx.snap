// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea render with snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root form__textarea css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    data-testid="textarea"
  >
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <textarea
        aria-invalid="false"
        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
        id=":r2:"
        placeholder="Введите сообщение"
        rows="4"
        style="height: 0px; overflow: hidden;"
      />
      <textarea
        aria-hidden="true"
        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
        readonly=""
        style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
        tabindex="-1"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-hdw1oc"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
